### PR TITLE
[REVIEW] Add INT8 to _schema_to_dtype, test

### LINKED ISSUE #1380: [BUG] Int8 not supported in GPUArrowReader
**Describe the bug**
For the OmniSci `select_ipc_gpu` method in pymapd, we use Arrow to return a buffer and use `GpuArrowReader` from cudf to build a GPU dataframe. Currently, having an `int8` column returns a `NotImplemented` error when trying to convert the Arrow buffer to a cudf.

I'm fairly certain the issue is here, and that adding `'INT8'` to the tuple on line 46 will fix the problem:
https://github.com/rapidsai/cudf/blob/a65a474d25c0c70e43fcc8d166264ff329243dcd/python/cudf/comm/gpuarrow.py#L43-L53

**Steps/Code to reproduce bug**
Need to be running an OmniSci server with GPU capabilities:

```python
import numpy as np
import pandas as pd
import pymapd

int8arr = np.array([1,2,3,4,5], dtype='int8')
int8df = pd.DataFrame(int8arr, columns=['int8col'])

con = pymapd.connect(host="localhost", port=6274, dbname="mapd", user="mapd", password="HyperInteractive")
con.load_table("int_table_bug", int8df, method="arrow")

gdf = con.select_ipc_gpu("select * from int_table_bug")

In [4]: gdf = con.select_ipc_gpu("select * from int_table_bug")
Traceback (most recent call last):

  File "<ipython-input-4-29f7c516646a>", line 1, in <module>
    gdf = con.select_ipc_gpu("select * from int_table_bug")

  File "/home/rzwitch/pymapd/pymapd/connection.py", line 278, in select_ipc_gpu
    df = _parse_tdf_gpu(tdf)

  File "/home/rzwitch/pymapd/pymapd/_parsers.py", line 206, in _parse_tdf_gpu
    reader = GpuArrowReader(schema_buffer, darr)

  File "/home/rzwitch/miniconda3/envs/cudf06/lib/python3.6/site-packages/cudf/comm/gpuarrow.py", line 175, in __init__
    self._open()

  File "/home/rzwitch/miniconda3/envs/cudf06/lib/python3.6/site-packages/cudf/comm/gpuarrow.py", line 209, in _open
    dtype=_schema_to_dtype(**layout['dtype']),

  File "/home/rzwitch/miniconda3/envs/cudf06/lib/python3.6/site-packages/cudf/comm/gpuarrow.py", line 52, in _schema_to_dtype
    raise NotImplementedError(fmt.format(name, bitwidth))

NotImplementedError: unsupported type INT8 8-bits

```

**Expected behavior**
The expectation is that a cudf can be created from an Arrow buffer with an `int8` column, just like it can when using `cudf.DataFrame.from_pandas()` where the pandas dataframe has an `int8` column.

**Environment details (please complete the following information):**
cudf 0.6, bare metal (home workstation) on Ubuntu 16.04 LTS

### PR DESCRIPTION
**Pull request by randyzwitch**
Closes #1380 

This PR adds `'INT8'` to the  `_schema_to_dtype` function, allowing `INT8` columns to be created in a DataFrame when coming from an Arrow RecordBatch.

In the tests, I parameterized the original `INT16` test to test all variants of `INT{8,16,32,64}` and removed the Arrow check for 0.7.1, as the minimum for this repo is now well beyond that point.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
ok to test

**Comment by kkraus14** [comment]
@randyzwitch looks like there's some flake8 issues:
```
>>>> FAILED: flake8 style check; begin output

python/cudf/tests/test_gpu_arrow_parser.py:166:69: F821 undefined name 'dtype'
python/cudf/tests/test_gpu_arrow_parser.py:167:71: F821 undefined name 'dtype'
python/cudf/tests/test_gpu_arrow_parser.py:185:41: F821 undefined name 'dtype'
```

And you need to add a line to `CHANGELOG.md` with a one-liner about this PR for the `check-changelog` as well.

File: python/cudf/tests/test_gpu_arrow_parser.py [hunk 1]
```diff
@@ -158,23 +158,18 @@ def test_gpu_parse_arrow_cats():
         np.testing.assert_almost_equal(got_weight, exp_weight)
 
 
-def make_gpu_parse_arrow_int16_batch():
-    depdelay = np.array([0, 0, -3, -2, 11, 6, -7, -4, 4, -3], dtype=np.int16)
-    arrdelay = np.array([5, -3, 1, -2, 22, 11, -12, -5, 4, -9], dtype=np.int16)
+@pytest.mark.skipif(arrow_version is None,
+                    reason='need compatible pyarrow to generate test data')
+@pytest.mark.parametrize('dtype', [np.int8, np.int16, np.int32, np.int64])
+def test_gpu_parse_arrow_int():
```
Comment (kkraus14) [inline python/cudf/tests/test_gpu_arrow_parser.py]: Looks like you forgot to pass the `dtype` as a parameter here so flake8 complains.

Comment (kkraus14) [reply python/cudf/tests/test_gpu_arrow_parser.py]:
```suggestion
def test_gpu_parse_arrow_int(dtype):
```

**Pull request merged**